### ISSUE TITLE
Ideas for covering more use cases

### ISSUE THREAD

**Issue by namannehra** [jamiebuilds/unstated-next#87]
## Problems

I am using unstated-next in a project. I found some situations where I couldn't use unstated-next and had to write contexts, providers and hooks myself. These were caused by following limitation of unstated-next.

### 1. Only initial state
Only initial state can be passed to providers. This makes it impossible to use unstated-next in certain situations. Initial state can be easily implemented in the function passed to `createContainer` using `useState`. So I think providers should always pass the state directly.

### 2. Only one prop on provider
Providers only accepts one prop named `initialState`. Making this configurable will be helpful.

### JS
```jsx
import React, { useState } from 'react'
import { createContainer } from 'unstated-next'

const useTheme = options => {
    const [initialType] = useState(options.initialType)
    const { primaryColor } = options
    // ...
}

const ThemeContainer = createContainer(useTheme)

const App = () => {
    <ThemeContainer.Provider initialType="light" primaryColor="red"></ThemeContainer.Provider>
}
```

### TS
```tsx
import React, { useState } from 'react'
import { createContainer } from 'unstated-next'

interface UseThemeOptions {
    initialType: 'light' | 'dark'
    primaryColor: string
}

const useTheme = (options: UseThemeOptions) => {
    const [initialType] = useState(options.initialType)
    const { primaryColor } = options
    // ...
}

const ThemeContainer = createContainer(useTheme)

const App = () => {
    <ThemeContainer.Provider initialType="light" primaryColor="red"></ThemeContainer.Provider>
}
```

## Notes

The type of the first argument can be used to determine the props of the provider. The provider can pass its props object directly to useContainer.
```ts
const createContainer = <T extends {}, U>(useContainer: (T) => U) => {
    type ProviderProps = T
    // ...
}
```

**Comment by namannehra** [comment]
@jamiebuilds Any thoughts on this?

**Comment by marcusmolchany** [comment]
I agree it would be really nice if we could pass props to a provider rather than only an `initialState` prop. My specific example is a provider with a smaller scope, such as a user profile on site. I want to change the user profile id globally, but the contents of the site are scoped to a lower component, and ideally I would just pass the `profileId` to the nested provider.
